src/services/state-persistor/state-persistor-layer.mapper.ts

```diff
@@ -0,0 +1,47 @@
+import { Layer } from '@/stores/map.store.model'
+import useLayers from '@/composables/layers/layers.composable'
+import useThemes from '@/composables/themes/themes.composable'
+import { BLANK_BACKGROUNDLAYER } from '@/composables/background-layer/background-layer.model'
+import { remoteLayerIdtoLayer } from '@/components/remote-layers/remote-layers.mapper'
+import { remoteLayersService } from '@/components/remote-layers/remote-layers.service'
+
+const STORAGE_SEPARATOR = '-'
+
+class LayerMapperService {
+  layerIdsToLayers(layerIdsText: string | null) {
+    const themes = useThemes()
+    const layers = useLayers()
+    const layerIds = layerIdsText
+      ? LayerMapperService.storageToLayerIds(layerIdsText)
+      : []
+
+    return layerIds.map(layerId => {
+      const layer = remoteLayersService.isRemoteLayer(layerId)
+        ? remoteLayerIdtoLayer(layerId)
+        : (themes.findById(parseInt(layerId, 10)) as unknown as Layer)
+
+      return layer ? layers.initLayer(layer) : void 0
+    })
+  }
+
+  layersToLayerIds(layers: Layer[] | null): string {
+    return layers?.map(layer => layer.id).join(STORAGE_SEPARATOR) || ''
+  }
+
+  bgLayerIdToBgLayer(bgLayerIdText: string | null): Layer | null {
+    const themes = useThemes()
+    return bgLayerIdText
+      ? (themes.findBgLayerByName(bgLayerIdText) as Layer)
+      : null
+  }
+
+  bgLayerTobgLayerId(layer: Layer | undefined) {
+    return layer?.name || BLANK_BACKGROUNDLAYER.name
+  }
+
+  static storageToLayerIds(layers: string): string[] {
+    return layers.split(STORAGE_SEPARATOR)
+  }
+}
+
+export const layerMapperService = new LayerMapperService()
```

Comment:
Not a big thing here, but I think we can improve a bit the naming.
`layerMapperService` is to vague.
Also, as the function names  in there are quite explicit, you can have maybe a more generic name as `StorageMapper`, or `StorageLayerMapper`